app/controllers/api/v0/base_controller.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module V0
+    class BaseController < ApplicationController
+      def list_order_item
+        item = OrderItem.where('id = ? and order_id = ?',
+                               params['order_item_id'], params['order_id']).first
+        render json: item.to_hash
+      end
+
+      def list_order_items
+        render json: OrderItem.where(:order_id => params['order_id']).collect(&:to_hash)
+      end
+
+      def list_orders
+        render json: Order.all.collect(&:to_hash)
+      end
+
+      def list_portfolios
+        portfolios = Portfolio.all
+        render json: portfolios
+      end
+
+      def list_portfolio_items
+        render json: PortfolioItem.all
+      end
+
+      def fetch_portfolio_with_id
+        render json: Portfolio.find_by(:id => params[:portfolio_id])
+      end
+
+      def fetch_portfolio_item_from_portfolio
+        items = Portfolio.find(params[:portfolio_id])
```

Comment:
No need to load the `Portfolio` here.  `PortfolioItem.find(params.require(:portfolio_item_id))`